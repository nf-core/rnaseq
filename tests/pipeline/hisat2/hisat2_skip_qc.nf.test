nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    options "-profile test,docker"
    tag "pipeline"

    test("HISAT2 (Skip QC) (GSE110004)") {

        when {
            params {
                outdir  = "$outputDir"
                aligner = "hisat2"
                skip_qc = true
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(path("$outputDir/pipeline_info/software_versions.yml")).match("software_versions") },
                { assert snapshot(path("$outputDir/pipeline_info/samplesheet.valid.csv")).match("samplesheet") },
                { assert new File("$outputDir/bbsplit/RAP1_IAA_30M_REP1.stats.txt").exists() },
                { assert new File("$outputDir/bbsplit/RAP1_UNINDUCED_REP1.stats.txt").exists() },
                { assert new File("$outputDir/bbsplit/RAP1_UNINDUCED_REP2.stats.txt").exists() },
                { assert new File("$outputDir/bbsplit/WT_REP1.stats.txt").exists() },
                { assert new File("$outputDir/bbsplit/WT_REP2.stats.txt").exists() },
                { assert !new File("$outputDir/fastqc/RAP1_IAA_30M_REP1_1_fastqc.html").exists() },
                { assert !new File("$outputDir/fastqc/RAP1_IAA_30M_REP1_2_fastqc.html").exists() },
                { assert !new File("$outputDir/fastqc/RAP1_UNINDUCED_REP1_fastqc.html").exists() },
                { assert !new File("$outputDir/fastqc/RAP1_UNINDUCED_REP2_fastqc.html").exists() },
                { assert !new File("$outputDir/fastqc/WT_REP1_1_fastqc.html").exists() },
                { assert !new File("$outputDir/fastqc/WT_REP1_2_fastqc.html").exists() },
                { assert !new File("$outputDir/fastqc/WT_REP2_1_fastqc.html").exists() },
                { assert !new File("$outputDir/fastqc/WT_REP2_2_fastqc.html").exists() },
                { assert new File("$outputDir/hisat2/bigwig/RAP1_IAA_30M_REP1.forward.bigWig").exists() },
                { assert new File("$outputDir/hisat2/bigwig/RAP1_IAA_30M_REP1.reverse.bigWig").exists() },
                { assert new File("$outputDir/hisat2/bigwig/RAP1_UNINDUCED_REP1.forward.bigWig").exists() },
                { assert new File("$outputDir/hisat2/bigwig/RAP1_UNINDUCED_REP1.reverse.bigWig").exists() },
                { assert new File("$outputDir/hisat2/bigwig/RAP1_UNINDUCED_REP2.forward.bigWig").exists() },
                { assert new File("$outputDir/hisat2/bigwig/RAP1_UNINDUCED_REP2.reverse.bigWig").exists() },
                { assert new File("$outputDir/hisat2/bigwig/WT_REP1.forward.bigWig").exists() },
                { assert new File("$outputDir/hisat2/bigwig/WT_REP1.reverse.bigWig").exists() },
                { assert new File("$outputDir/hisat2/bigwig/WT_REP2.forward.bigWig").exists() },
                { assert new File("$outputDir/hisat2/bigwig/WT_REP2.reverse.bigWig").exists() },
                { assert new File("$outputDir/hisat2/log/RAP1_IAA_30M_REP1.hisat2.summary.log").exists() },
                { assert new File("$outputDir/hisat2/log/RAP1_UNINDUCED_REP1.hisat2.summary.log").exists() },
                { assert new File("$outputDir/hisat2/log/RAP1_UNINDUCED_REP2.hisat2.summary.log").exists() },
                { assert new File("$outputDir/hisat2/log/WT_REP1.hisat2.summary.log").exists() },
                { assert new File("$outputDir/hisat2/log/WT_REP2.hisat2.summary.log").exists() },
                { assert new File("$outputDir/hisat2/samtools_stats/RAP1_IAA_30M_REP1.markdup.sorted.bam.flagstat").exists() },
                { assert new File("$outputDir/hisat2/samtools_stats/RAP1_IAA_30M_REP1.markdup.sorted.bam.idxstats").exists() },
                { assert new File("$outputDir/hisat2/samtools_stats/RAP1_IAA_30M_REP1.markdup.sorted.bam.stats").exists() },
                { assert new File("$outputDir/hisat2/samtools_stats/RAP1_IAA_30M_REP1.sorted.bam.flagstat").exists() },
                { assert new File("$outputDir/hisat2/samtools_stats/RAP1_IAA_30M_REP1.sorted.bam.idxstats").exists() },
                { assert new File("$outputDir/hisat2/samtools_stats/RAP1_IAA_30M_REP1.sorted.bam.stats").exists() },
                { assert new File("$outputDir/hisat2/samtools_stats/RAP1_UNINDUCED_REP1.markdup.sorted.bam.flagstat").exists() },
                { assert new File("$outputDir/hisat2/samtools_stats/RAP1_UNINDUCED_REP1.markdup.sorted.bam.idxstats").exists() },
                { assert new File("$outputDir/hisat2/samtools_stats/RAP1_UNINDUCED_REP1.markdup.sorted.bam.stats").exists() },
                { assert new File("$outputDir/hisat2/samtools_stats/RAP1_UNINDUCED_REP1.sorted.bam.flagstat").exists() },
                { assert new File("$outputDir/hisat2/samtools_stats/RAP1_UNINDUCED_REP1.sorted.bam.idxstats").exists() },
                { assert new File("$outputDir/hisat2/samtools_stats/RAP1_UNINDUCED_REP1.sorted.bam.stats").exists() },
                { assert new File("$outputDir/hisat2/samtools_stats/RAP1_UNINDUCED_REP2.markdup.sorted.bam.flagstat").exists() },
                { assert new File("$outputDir/hisat2/samtools_stats/RAP1_UNINDUCED_REP2.markdup.sorted.bam.idxstats").exists() },
                { assert new File("$outputDir/hisat2/samtools_stats/RAP1_UNINDUCED_REP2.markdup.sorted.bam.stats").exists() },
                { assert new File("$outputDir/hisat2/samtools_stats/RAP1_UNINDUCED_REP2.sorted.bam.flagstat").exists() },
                { assert new File("$outputDir/hisat2/samtools_stats/RAP1_UNINDUCED_REP2.sorted.bam.idxstats").exists() },
                { assert new File("$outputDir/hisat2/samtools_stats/RAP1_UNINDUCED_REP2.sorted.bam.stats").exists() },
                { assert new File("$outputDir/hisat2/samtools_stats/WT_REP1.markdup.sorted.bam.flagstat").exists() },
                { assert new File("$outputDir/hisat2/samtools_stats/WT_REP1.markdup.sorted.bam.idxstats").exists() },
                { assert new File("$outputDir/hisat2/samtools_stats/WT_REP1.markdup.sorted.bam.stats").exists() },
                { assert new File("$outputDir/hisat2/samtools_stats/WT_REP1.sorted.bam.flagstat").exists() },
                { assert new File("$outputDir/hisat2/samtools_stats/WT_REP1.sorted.bam.idxstats").exists() },
                { assert new File("$outputDir/hisat2/samtools_stats/WT_REP1.sorted.bam.stats").exists() },
                { assert new File("$outputDir/hisat2/samtools_stats/WT_REP2.markdup.sorted.bam.flagstat").exists() },
                { assert new File("$outputDir/hisat2/samtools_stats/WT_REP2.markdup.sorted.bam.idxstats").exists() },
                { assert new File("$outputDir/hisat2/samtools_stats/WT_REP2.markdup.sorted.bam.stats").exists() },
                { assert new File("$outputDir/hisat2/samtools_stats/WT_REP2.sorted.bam.flagstat").exists() },
                { assert new File("$outputDir/hisat2/samtools_stats/WT_REP2.sorted.bam.idxstats").exists() },
                { assert new File("$outputDir/hisat2/samtools_stats/WT_REP2.sorted.bam.stats").exists() },
                { assert snapshot(path("$outputDir/multiqc/hisat2/multiqc_data/multiqc_cutadapt.txt"),
                                path("$outputDir/multiqc/hisat2/multiqc_data/multiqc_fastqc.txt"),
                                path("$outputDir/multiqc/hisat2/multiqc_data/multiqc_picard_dups.txt"),
                                path("$outputDir/multiqc/hisat2/multiqc_data/picard_histogram.txt")).match("multiqc") },
                { assert new File("$outputDir/multiqc/hisat2/multiqc_data/multiqc_hisat2.txt").exists() },
                { assert new File("$outputDir/multiqc/hisat2/multiqc_data/multiqc_general_stats.txt").exists() },
                { assert new File("$outputDir/multiqc/hisat2/multiqc_data/multiqc_samtools_flagstat.txt").exists() },
                { assert new File("$outputDir/multiqc/hisat2/multiqc_data/multiqc_samtools_idxstats.txt").exists() },
                { assert new File("$outputDir/multiqc/hisat2/multiqc_data/multiqc_samtools_stats.txt").exists() },
                { assert snapshot(path("$outputDir/salmon/salmon_tx2gene.tsv")).match("salmon") },
                { assert new File("$outputDir/salmon/salmon.merged.gene_counts.tsv").exists() },
                { assert new File("$outputDir/salmon/salmon.merged.transcript_counts.tsv").exists() },
                { assert new File("$outputDir/salmon/RAP1_IAA_30M_REP1/quant.sf").exists() },
                { assert new File("$outputDir/salmon/RAP1_IAA_30M_REP1/quant.genes.sf").exists() },
                { assert new File("$outputDir/salmon/RAP1_UNINDUCED_REP1/quant.sf").exists() },
                { assert new File("$outputDir/salmon/RAP1_UNINDUCED_REP1/quant.genes.sf").exists() },
                { assert new File("$outputDir/salmon/RAP1_UNINDUCED_REP2/quant.sf").exists() },
                { assert new File("$outputDir/salmon/RAP1_UNINDUCED_REP2/quant.genes.sf").exists() },
                { assert new File("$outputDir/salmon/WT_REP1/quant.sf").exists() },
                { assert new File("$outputDir/salmon/WT_REP1/quant.genes.sf").exists() },
                { assert new File("$outputDir/salmon/WT_REP2/quant.sf").exists() },
                { assert new File("$outputDir/salmon/WT_REP2/quant.genes.sf").exists() },
                { assert new File("$outputDir/hisat2/picard_metrics/RAP1_IAA_30M_REP1.markdup.sorted.MarkDuplicates.metrics.txt").exists() },
                { assert new File("$outputDir/hisat2/picard_metrics/RAP1_UNINDUCED_REP1.markdup.sorted.MarkDuplicates.metrics.txt").exists() },
                { assert new File("$outputDir/hisat2/picard_metrics/RAP1_UNINDUCED_REP2.markdup.sorted.MarkDuplicates.metrics.txt").exists() },
                { assert new File("$outputDir/hisat2/picard_metrics/WT_REP1.markdup.sorted.MarkDuplicates.metrics.txt").exists() },
                { assert new File("$outputDir/hisat2/picard_metrics/WT_REP2.markdup.sorted.MarkDuplicates.metrics.txt").exists() },
                { assert new File("$outputDir/hisat2/RAP1_IAA_30M_REP1.markdup.sorted.bam").exists() },
                { assert new File("$outputDir/hisat2/RAP1_IAA_30M_REP1.markdup.sorted.bam.bai").exists() },
                { assert new File("$outputDir/hisat2/WT_REP1.markdup.sorted.bam").exists() },
                { assert new File("$outputDir/hisat2/WT_REP1.markdup.sorted.bam.bai").exists() },
                { assert new File("$outputDir/hisat2/RAP1_UNINDUCED_REP1.markdup.sorted.bam").exists() },
                { assert new File("$outputDir/hisat2/RAP1_UNINDUCED_REP1.markdup.sorted.bam.bai").exists() },
                { assert new File("$outputDir/hisat2/WT_REP2.markdup.sorted.bam").exists() },
                { assert new File("$outputDir/hisat2/WT_REP2.markdup.sorted.bam.bai").exists() },
                { assert new File("$outputDir/hisat2/RAP1_UNINDUCED_REP2.markdup.sorted.bam").exists() },
                { assert new File("$outputDir/hisat2/RAP1_UNINDUCED_REP2.markdup.sorted.bam.bai").exists() },
                { assert new File("$outputDir/hisat2/stringtie/RAP1_IAA_30M_REP1.gene.abundance.txt").exists() },
                { assert new File("$outputDir/hisat2/stringtie/RAP1_UNINDUCED_REP1.gene.abundance.txt").exists() },
                { assert new File("$outputDir/hisat2/stringtie/RAP1_UNINDUCED_REP2.gene.abundance.txt").exists() },
                { assert new File("$outputDir/hisat2/stringtie/WT_REP1.gene.abundance.txt").exists() },
                { assert new File("$outputDir/hisat2/stringtie/WT_REP2.gene.abundance.txt").exists() },
                { assert new File("$outputDir/hisat2/stringtie/RAP1_IAA_30M_REP1.coverage.gtf").exists() },
                { assert new File("$outputDir/hisat2/stringtie/RAP1_UNINDUCED_REP1.coverage.gtf").exists() },
                { assert new File("$outputDir/hisat2/stringtie/RAP1_UNINDUCED_REP2.coverage.gtf").exists() },
                { assert new File("$outputDir/hisat2/stringtie/WT_REP1.coverage.gtf").exists() },
                { assert new File("$outputDir/hisat2/stringtie/WT_REP2.coverage.gtf").exists() },
                { assert new File("$outputDir/hisat2/stringtie/RAP1_IAA_30M_REP1.transcripts.gtf").exists() },
                { assert new File("$outputDir/hisat2/stringtie/RAP1_UNINDUCED_REP1.transcripts.gtf").exists() },
                { assert new File("$outputDir/hisat2/stringtie/RAP1_UNINDUCED_REP2.transcripts.gtf").exists() },
                { assert new File("$outputDir/hisat2/stringtie/WT_REP1.transcripts.gtf").exists() },
                { assert new File("$outputDir/hisat2/stringtie/WT_REP2.transcripts.gtf").exists() },
                { assert new File("$outputDir/trimgalore/RAP1_IAA_30M_REP1_1.fastq.gz_trimming_report.txt").exists() },
                { assert new File("$outputDir/trimgalore/RAP1_IAA_30M_REP1_2.fastq.gz_trimming_report.txt").exists() },
                { assert new File("$outputDir/trimgalore/RAP1_UNINDUCED_REP1.fastq.gz_trimming_report.txt").exists() },
                { assert new File("$outputDir/trimgalore/RAP1_UNINDUCED_REP2.fastq.gz_trimming_report.txt").exists() },
                { assert new File("$outputDir/trimgalore/WT_REP1_1.fastq.gz_trimming_report.txt").exists() },
                { assert new File("$outputDir/trimgalore/WT_REP1_2.fastq.gz_trimming_report.txt").exists() },
                { assert new File("$outputDir/trimgalore/WT_REP2_1.fastq.gz_trimming_report.txt").exists() },
                { assert new File("$outputDir/trimgalore/WT_REP2_2.fastq.gz_trimming_report.txt").exists() }
            )
        }

    }

}
