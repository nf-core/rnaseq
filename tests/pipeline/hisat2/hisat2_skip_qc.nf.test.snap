{
    "samplesheet": {
        "content": [
            "samplesheet.valid.csv:md5,3d40bf3d424290cec0dbe9ee95b7dd12"
        ],
        "timestamp": "2023-05-08T18:31:28+0000"
    },
    "software_versions": {
        "content": [
            "software_versions.yml:md5,3ef1180b059398ea06240e301111ff4d"
        ],
        "timestamp": "2023-05-08T18:31:28+0000"
    },
    "salmon": {
        "content": [
            "salmon_tx2gene.tsv:md5,b367acfc4b7df7fb68fb1b1d2956e27e"
        ],
        "timestamp": "2023-05-08T18:31:28+0000"
    },
    "multiqc": {
        "content": [
            "multiqc_cutadapt.txt:md5,aac9581a5670cb55edf564f3d6c1f9a7",
            "multiqc_fastqc.txt:md5,e3ebd5c2676bebb3c0176c4e985f3580",
            "multiqc_picard_dups.txt:md5,71d5bafee1a4916951349018f387f7a0",
            "picard_histogram.txt:md5,c9aa8a5ac6841ffb3a6cc2de45b44797"
        ],
        "timestamp": "2023-05-08T18:31:28+0000"
    }
}